docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,351 @@
+
+### <a name='ImplementationDetailsNotesConstraints'></a>Implementation Details/Notes/Constraints
+
+- AAD pod identity can co exist with AWI
+- Migration plan from AAD pod identity to AWI for existing cluster. Refer to the `Migration Plan` section at the bottom of the document.
+- For AWI to work the following prerequisites must be met.
+  - Key Generation
+  - OIDC URL Setup
+  - Set Service Account Signing Flags
+
+#### <a name='KeyGeneration'></a>Key Generation
+
+Admin should generate signing key pairs by using a tool such as openssl or bring their own public and private keys. 
+These keys will be mounted on a path on the containers running on the control plane node. These keys are required for signing the service account tokens that will be used by the capz pod. 
+
+#### <a name='OidcUrlSetup'></a>OIDC URL Setup
+
+Two documents i.e OIDC and JWKS json documents needs to be generated and published to a public URL. The OIDC document contains the metadata of the issuer. The JSON Web Key Sets (JWKS) document contains the public signing key(s) that allows AAD to verify the authenticity of the service account token.
+
+For more details on how to set up OIDC URL and more details on it, please g through this this [link](#https://azure.github.io/azure-workload-identity/docs/installation/self-managed-clusters/oidc-issuer.html)
+
+The steps on a high level to setup is the following
+- Create an azure blob storage account.
+- Create a storage container.
+- Generate the OIDC and JWKS document.
+- The doument should be accessible on the public accessible URL which will be used later.
+
+#### <a name='SetServiceAccountSigningFlags'></a>Set Service Account Signing Flags
+
+Setup the flags on the kind cluster. An example is shown below
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+      # path on node where the public key exists
+    - hostPath: ${SERVICE_ACCOUNT_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.pub
+	  # path on node where the private key exists
+    - hostPath: ${SERVICE_ACCOUNT_SIGNING_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.key
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+	    # the oidc url after it has been set up
+        service-account-issuer: ${SERVICE_ACCOUNT_ISSUER}
+        service-account-key-file: /etc/kubernetes/pki/sa.pub
+        service-account-signing-key-file: /etc/kubernetes/pki/sa.key
+    controllerManager:
+      extraArgs:
+        service-account-private-key-file: /etc/kubernetes/pki/sa.key
+```
+
+#### <a name='FederatedCredential'></a>Federated Credential
+
+The service account that the capz pod will use should be annotated with the client ID of the AAD application or User Assigned Identity and a fedrated identity should be created by associating the ServiceAccont as a subject.
```

Comment:
```suggestion
The service account that the capz pod will use should be annotated with the client ID of the AAD application or user-sssigned Identity and a federated identity should be created by associating the Service Account as a subject.
```

Reply:
Again in this case, the annotated part is optional. So I will remove this.
Infact, in the entire document, I am planning to remove this ''annotated part'' and mention it as a way to fallback to use client ID if it is no where supplied. 
Does this make sense?

Reply:
Where else could the client ID be specified? Would it be possible to provide the `client-id` and `federated-token-file` path as part of the existing credentials secret (while omitting the `client-secret` for example with AZWI) as an alternative to injecting environment variables via the mutating webhook? https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/config/default/credentials.yaml